src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs

```diff
@@ -414,6 +414,252 @@ public static class C
+    }
+}
+";
+
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            compilation.VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput:
+@"param1: param1_default, param2: param2_default
+param1: param1_default, param2: param2_default
+param1: param1_value, param2: ""param1_value""
+param1: param1_value, param2: ""param1_value""
+param1: ""param2_value"", param2: param2_value
+param1: param1_value, param2: param2_value
+param1: param1_value, param2: param2_value");
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestArgumentExpressionInAttributeConstructor_NamedAndOptionalParameters()
+        {
+            string source = @"
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+class MyAttribute : Attribute
+{
+    public MyAttribute(int a = 1, int b = 2, int c = 3, [CallerArgumentExpression(""a"")] string expr_a = """", [CallerArgumentExpression(""b"")] string expr_b = """", [CallerArgumentExpression(""c"")] string expr_c = """")
+    {
+        Console.WriteLine($""'{a}', '{b}', '{c}', '{expr_a}', '{expr_b}', '{expr_c}'"");
+        A = a;
+        B = b;
+        C = c;
+    }
+
+    public int X;
+    public int A;
+    public int B;
+    public int C;
+}
+
+[My(0+0, c:1+1, X=2+2)]
+class Program
+{
+    static void Main()
+    {
+        typeof(Program).GetCustomAttribute(typeof(MyAttribute));
+        _ = new MyAttribute(0+0, c: 1+1);
+    }
+}";
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            compilation.VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: @"'0', '2', '2', '0+0', '', '1+1'
+'0', '2', '2', '0+0', '', '1+1'");
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestArgumentExpressionInAttributeConstructor_OptionalAndFieldInitializer()
+        {
+            string source = @"
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+class MyAttribute : Attribute
+{
+    public MyAttribute([CallerArgumentExpression(""a"")] string expr_a = ""<default0>"", string a = ""<default1>"")
+    {
+        Console.WriteLine($""'{a}', '{expr_a}'"");
+    }
+
+    public int A;
+    public int B;
+}
+
+[My(A=1, B=2)]
+class Program
+{
+    static void Main()
+    {
+        typeof(Program).GetCustomAttribute(typeof(MyAttribute));
+    }
+}";
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            compilation.VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "'<default1>', '<default0>'");
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Whatever scenarios we tested for regular invocations, we probably should make sure we cover them for attributes as well.

Reply:
> Whatever scenarios we tested for regular invocations, we probably should make sure we cover them for attributes as well.

Added tests (which revealed few bugs)

Reply:
> Added tests (which revealed few bugs)

Could you elaborate please? I do not see any obvious fixes in the compiler.

Reply:
@AlekseyTs This test isn't yet fixed. The bug is still there and I've a PROTOTYPE for it. The test is `TestCallerArgumentWithMemberNameAttributes_AttributeConstructor` which isn't consistent with invocations (it has an extra error).

(I originally said tests, the other one wasn't a product bug, but a test bug - so it's only the one test I stated above)